next.config.js

```diff
@@ -11,20 +11,57 @@ module.exports = withBundleAnalyzer({
     polyfillsOptimization: true,
     jsconfigPaths: true,
   },
-  serverRuntimeConfig: {
+  publicRuntimeConfig: {
     MOCKNET_API_SERVER: process.env.MOCKNET_API_SERVER,
     TESTNET_API_SERVER: process.env.TESTNET_API_SERVER,
     STAGING: process.env.STAGING,
   },
   webpack(config, { dev }) {
+    config.output.webassemblyModuleFilename = 'static/wasm/[modulehash].wasm';
+
+    // Install webpack aliases:
+    const aliases = config.resolve.alias || (config.resolve.alias = {});
+
     if (!dev) {
-      config.externals.push('elliptic');
-      config.plugins.push(new webpack.IgnorePlugin(/^\.\/wordlists\/(?!english)/, /bip39\/src$/));
+      config.plugins.push(
+        new webpack.IgnorePlugin({
+          resourceRegExp: /^\.\/wordlists\/(?!english)/,
+          contextRegExp: /bip39\/src$/,
+        })
+      );
+
+      const splitChunks = config.optimization && config.optimization.splitChunks;
+
+      if (splitChunks) {
+        const cacheGroups = splitChunks.cacheGroups;
+        const test = /[\\/]node_modules[\\/](preact|preact-render-to-string|preact-context-provider)[\\/]/;
+        if (cacheGroups.framework) {
+          cacheGroups.preact = Object.assign({}, cacheGroups.framework, {
+            test,
+          });
+          cacheGroups.commons.name = 'framework';
+        } else {
+          cacheGroups.preact = {
+            name: 'commons',
+            chunks: 'all',
+            test,
+          };
+        }
+      }
```

Comment:
pulls out preact into the framework bundle, for better bundle size and dedupe. doesn't automatically happen because next.js is made for react. see https://github.com/preactjs/next-plugin-preact/blob/master/packages/next-plugin-preact/index.js#L29